bin/aws_zsh_completer.sh

```diff
@@ -1,60 +1,79 @@
 
-autoload -Uz bashcompinit
-bashcompinit -i
+# Load and initialize the completion system ignoring insecure directories.
+autoload -Uz bashcompinit && bashcompinit -i
 
-_bash_complete() {
-  local ret=1
-  local -a suf matches
-  local -x COMP_POINT COMP_CWORD
-  local -a COMP_WORDS COMPREPLY BASH_VERSINFO
-  local -x COMP_LINE="$words"
-  local -A savejobstates savejobtexts
+# Overwrite _bash_complete() for zsh versions less than 5.0.3 only
+autoload -Uz is-at-least
+if ! is-at-least 5.0.3; then
+  _bash_complete() {
+    local ret=1
+    local -a suf matches
+    local -x COMP_POINT COMP_CWORD
+    local -a COMP_WORDS COMPREPLY BASH_VERSINFO
+    local -x COMP_LINE="$words"
+    local -A savejobstates savejobtexts
 
-  (( COMP_POINT = 1 + ${#${(j. .)words[1,CURRENT]}} + $#QIPREFIX + $#IPREFIX + $#PREFIX ))
-  (( COMP_CWORD = CURRENT - 1))
-  COMP_WORDS=( $words )
-  BASH_VERSINFO=( 2 05b 0 1 release )
+    (( COMP_POINT = 1 + ${#${(j. .)words[1,CURRENT]}} + $#QIPREFIX + $#IPREFIX + $#PREFIX ))
+    (( COMP_CWORD = CURRENT - 1))
+    COMP_WORDS=( $words )
+    BASH_VERSINFO=( 2 05b 0 1 release )
 
-  savejobstates=( ${(kv)jobstates} )
-  savejobtexts=( ${(kv)jobtexts} )
+    savejobstates=( ${(kv)jobstates} )
+    savejobtexts=( ${(kv)jobtexts} )
 
-  [[ ${argv[${argv[(I)nospace]:-0}-1]} = -o ]] && suf=( -S '' )
+    [[ ${argv[${argv[(I)nospace]:-0}-1]} = -o ]] && suf=( -S '' )
 
-  matches=( ${(f)"$(compgen $@ -- ${words[CURRENT]})"} )
+    matches=( ${(f)"$(compgen $@ -- ${words[CURRENT]})"} )
 
-  if [[ -n $matches ]]; then
-    if [[ ${argv[${argv[(I)filenames]:-0}-1]} = -o ]]; then
-      compset -P '*/' && matches=( ${matches##*/} )
-      compset -S '/*' && matches=( ${matches%%/*} )
-      compadd -Q -f "${suf[@]}" -a matches && ret=0
-    else
-      compadd -Q "${suf[@]}" -a matches && ret=0
+    if [[ -n $matches ]]; then
+      if [[ ${argv[${argv[(I)filenames]:-0}-1]} = -o ]]; then
+        compset -P '*/' && matches=( ${matches##*/} )
+        compset -S '/*' && matches=( ${matches%%/*} )
+        compadd -Q -f "${suf[@]}" -a matches && ret=0
+      else
+        compadd -Q "${suf[@]}" -a matches && ret=0
+      fi
     fi
-  fi
 
-  if (( ret )); then
-    if [[ ${argv[${argv[(I)default]:-0}-1]} = -o ]]; then
-      _default "${suf[@]}" && ret=0
-    elif [[ ${argv[${argv[(I)dirnames]:-0}-1]} = -o ]]; then
-      _directories "${suf[@]}" && ret=0
+    if (( ret )); then
+      if [[ ${argv[${argv[(I)default]:-0}-1]} = -o ]]; then
+        _default "${suf[@]}" && ret=0
+      elif [[ ${argv[${argv[(I)dirnames]:-0}-1]} = -o ]]; then
+        _directories "${suf[@]}" && ret=0
+      fi
     fi
-  fi
 
-  return ret
-}
+    return ret
+  }
+fi
 
 complete -C aws_completer aws
+
+# Bind `aws` completion eagerly to `_bash_complete -C aws_completer` and
+# suppress stderr to avoid 'can only be called from completion function' warning
+_complete aws # 2> /dev/null
```

Comment:
```suggestion
_complete aws 2> /dev/null
```
An extra change is needed to actually get rid of 'can only be called from completion function' warning